packages/relay/src/lib/eth.ts

```diff
@@ -624,15 +657,31 @@ export class EthImpl implements Eth {
     const latestBlockTolerance = 1;
     this.logger.trace(`${requestIdPrefix} getBalance(account=${account}, blockNumberOrTag=${blockNumberOrTag})`);
 
+    let latestBlock: LatestBlockNumberTimestamp | null | undefined;
     // this check is required, because some tools like Metamask pass for parameter latest block, with a number (ex 0x30ea)
     // tolerance is needed, because there is a small delay between requesting latest block from blockNumber and passing it here
     if (!EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
-      const latestBlock = await this.blockNumber(requestId);
-      const blockDiff = Number(latestBlock) - Number(blockNumberOrTag);
+      const cacheKey = `${constants.CACHE_KEY.ETH_BLOCK_NUMBER}`;
```

Comment:
are we caching a single block value?  It seems to me that the cache key is a constant `'eth_block_number'`.  Do we not want to add the block number to the cache key somehow?

Reply:
Yes, we're only caching a single block value with a time to live set to 1000 ms.  This snippet was extracted from the original blockNumber method that's been working up to now.  The problem was the blockNumber method which is used in many places only returns the block number and later in the getBalance process the latest block timestamp is needed which caused another call to the mirror node.  In this new getBalance only version of the method we also return the latest block timestamp saving a call to the mirrornode.

Reply:
So we are using the cache only for its ttl feature.  Makes sense